h/views/api/config.py

```diff
@@ -51,24 +63,45 @@ def add_api_view(
         # preferred
         primary_method = primary_method[0]
 
-    settings.setdefault("accept", "application/json")
     settings.setdefault("renderer", "json")
     settings.setdefault("decorator", cors_policy)
 
-    if link_name:
-        link = {
-            "name": link_name,
-            "method": primary_method,
-            "route_name": settings.get("route_name"),
-            "description": description,
-        }
-
-        registry = config.registry
-        if not hasattr(registry, "api_links"):
-            registry.api_links = []
-        registry.api_links.append(link)
+    current_version = config.registry.settings["api.version.current"]
+    known_versions = config.registry.settings["api.versions"]
+    view_versions = versions or [current_version]
+
+    registry = config.registry
+    if not hasattr(registry, "api_links"):
+        registry.api_links = {}
+
+    for version in view_versions:
+        if version not in known_versions:
+            raise APIConfigError(
+                "Version " + version + " does not match any known API version"
+            )
+        if version == current_version:
+            settings.setdefault("accept", "application/json")
+
+        # Register one link per version supported by this view
+        if link_name:
+            link = {
+                "name": link_name,
+                "method": primary_method,
+                "route_name": settings.get("route_name"),
+                "description": description,
+            }
+            if version not in registry.api_links:
+                registry.api_links[version] = []
+            registry.api_links[version].append(link)
 
     config.add_view(view=view, **settings)
+
+    # Add version-specific Accept headers. This requires re-adding the view
+    # as `add_view` only takes one value for `accept` at a time
+    for version in view_versions:
+        settings["accept"] = "application/vnd.hypothesis." + version + "+json"
+        config.add_view(view=view, **settings)
```

Comment:
Hokay. This was the most annoying part to figure out. Yeah, you an call `config.add_view` multiple times on the same view, but I can't just do `config.add_view(view=view, accept="relevant_accept_header"); I have to pass on the `settings` from the first call to `add_view` or it I get conflict errors. It's possible that only a subset of those `settings` are needed to extend an existing view's config, but it wasn't obvious what and I didn't want to take the time to futz around with that. So: looking for any input on this. Maybe it's fine as-is? It's sort of a Pyramid oddity that one cannot extend existing config, only "re-add" the view in entirety. Shrug.

Reply:
I'm not sure why Pyramid's `accept=` doesn't accept a list of strings :man_shrugging: But given that it doesn't I wonder whether the `api_version` setting should also be a single string not a list of strings?

Reply:
I want views to be able to support multiple versions (e.g. `versions=["v1", "v2"]`)...

Reply:
(I feel like I might be mis-reading your comment...can you rephrase, actually?)

Reply:
As in:

```python
@api_config(... version="v1")
@api_config(... version="v2")
def my_view(...):
    ...
```